python/descriptor.c

```diff
@@ -144,34 +145,33 @@ static PyObject* PyUpb_DescriptorBase_GetOptions(PyUpb_DescriptorBase* self,
 
 typedef void* PyUpb_ToProto_Func(const void* def, upb_arena* arena);
 
-static PyObject* PyUpb_DescriptorBase_CopyToProto(PyObject* _self,
-                                                  PyUpb_ToProto_Func* func,
-                                                  const upb_msglayout* layout,
-                                                  PyObject* py_proto) {
+static PyObject* PyUpb_DescriptorBase_GetSerializedProto(
+    PyObject* _self, PyUpb_ToProto_Func* func, const upb_msglayout* layout) {
   PyUpb_DescriptorBase* self = (void*)_self;
-  PyObject* ret = NULL;
-  PyObject* str = NULL;
   upb_arena* arena = upb_arena_new();
   if (!arena) PYUPB_RETURN_OOM;
   upb_msg* proto = func(self->def, arena);
   if (!proto) goto oom;
   size_t size;
   char* pb = upb_encode(proto, layout, arena, &size);
   if (!pb) goto oom;
-  str = PyBytes_FromStringAndSize(pb, size);
-  if (!str) goto oom;
-  // Disabled until python/message.c is reviewed and checked in.
-  // PyObject *ret = PyUpb_CMessage_MergeFromString(py_proto, str);
-  PyErr_Format(PyExc_NotImplementedError, "Not yet implemented");
-
-done:
-  Py_XDECREF(str);
+  PyObject* str = PyBytes_FromStringAndSize(pb, size);
   upb_arena_free(arena);
-  return ret;
+  return str;
 
 oom:
   PyErr_SetNone(PyExc_MemoryError);
-  goto done;
+  return NULL;
```

Comment:
pretty sure you are missing an arena free call on the oom path

Reply:
Done.